### ISSUE TITLE
FR: lower limit for rotation gestures

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#12118]
**Does your feature request involve difficulty completing a task? Please describe.**
Many times I trigger rotation gestures by accident when trying other gestures like two-finger swipe.

**Describe the solution you'd like**
Menu entry to set lower limit for rotation gestures, say, to 45 degrees.

I can see the angle is calculated (`9.14` degrees, I guess?):
```
06/28/24-13:35:17 DEBUG Contact:voidState Deferring to panState via buddy slot 0 to handle MT contact lift for a rotate 
06/28/24-13:35:17 DEBUG slot 0 in pan state... 
06/28/24-13:35:17 DEBUG Contact:handleTwoFingerPan: rotate -9.1492974387141 detected 
06/28/24-13:35:17 DEBUG rotate ccw 9.1492974387141 detected 
```

**Describe alternatives you've considered**
Make sure other gestures are straight, however, in a moving bus this is many times harder :D 

**Additional context**
The only reason I have those rotation gestures (with restart and exit) is to have a backup for times when a nightly screws up the input or something (usually in my rotation tests) :D 
Admittedly, this hasn't happened for a long time and I haven't worked on the rotation for a long time as well.

**Comment by Frenzie** [comment]
In https://github.com/koreader/koreader/pull/9463 it was added as "90°" which seems to imply having to do quite a lot, but indeed fairly minuscule rotation suffices. It does seem nearly impossible to trigger it unless you keep one finger still though?

**Comment by mergen3107** [comment]
> unless you keep one finger still though

That is correct. However, in a moving bus what is supposed to be "two moving fingers diagonally to the top left" might become "one finger moving away, while the other tries to stay on the screen" if the movement is quite sudden :D

**Comment by mergen3107** [comment]
So this `rotate ccw 9.1492974387141 detected` looks believable.
I was going to the top left with two fingers on my left hand to summon Page Browser, but it got treated as a counter-clockwise rotation.

**Comment by NiLuJe** [comment]
That's because what @Frenzie just described is the *only* discriminant between a two-finger pan/swipe and a rotate.

Quoting myself,

https://github.com/koreader/koreader/blob/4f75e4163626314bc4fc1b3193b7a5b6f026243d/frontend/device/gesturedetector.lua#L1076

See the comments above that for why I opted *not* to add more heuristics on top of that based on the detected angle.

TL;DR: This gesture is the bane of my existence. It's responsible for a good 60% of the jank in GestureDetector :/.

**Comment by Frenzie** [comment]
Has that rotate gesture always been there? I never use it. :-P

**Comment by NiLuJe** [comment]
As far back as I can remember, yeah ;).

I don't think it's bound to anything by default, which is sort of how we get away with it being super janky ;p.

**Comment by mergen3107** [comment]
> TL;DR: Both fingers need to move for a pinch/spread, while a finger needs **to stay still** for a rotate. 

@NiLuJe 
In real world, I think it is impossible to have one finder still-still (according to coordinates).

Is it quantified somewhere how much small movement is considered "still"?

**Comment by NiLuJe** [comment]
> Is it quantified somewhere how much small movement is considered "still"?

IIRC, below the pan threshold.

https://github.com/koreader/koreader/blob/ce7fcff3608cd3f4ee24898b82a3c2e680f202c3/frontend/device/gesturedetector.lua#L117

**Comment by mergen3107** [comment]
@NiLuJe 
How hard is it to make this `self.PAN_THRESHOLD` as Top Menu option?
I would love to play around with it live

**Comment by NiLuJe** [comment]
Probably not terribly so, since the interval parameters are already handled via user settings.
